sw/device/sca/lib/sca.h

```diff
@@ -165,6 +165,7 @@ void sca_call_and_sleep(sca_callee callee, uint32_t sleep_cycles);
  * TODO: Remove once there is a CHECK_DIF_OK version that does not require test
  * library dependencies.
  */
+#ifndef CHECK_DIF_OK
```

Comment:
Hm, is this really needed?

Reply:
Good question, I removed it now.

Previously, the `_ecc*_serial.c` files were using the `CHECK` macro from [sw/device/lib/testing/test_framework/check.h](https://github.com/lowRISC/opentitan/blob/master/sw/device/lib/testing/test_framework/check.h) to verify if a given condition is true:

 `#define CHECK(condition, ...)                        \`

As the  [sw/device/lib/testing/test_framework/check.h](https://github.com/lowRISC/opentitan/blob/master/sw/device/lib/testing/test_framework/check.h) also contains the definition of `CHECK_DIF_OK`, I was getting a compilation error.

I noticed that we already have a similar macro,`SS_CHECK`, in [sw/device/sca/lib/simple_serial.h](https://github.com/lowRISC/opentitan/blob/master/sw/device/sca/lib/simple_serial.h).  In the `ecc*_serial.c` files, I replaced all instances of `CHECK` with `SS_CHECK`, and removed `ifndef` statement from `sw/device/sca/lib/sca.h`:

`#define SS_CHECK(condition)                          \`

Reply:
Great, thanks for cleaning this up.